### ISSUE TITLE
recycling FIT nodes

### ISSUE THREAD

**Issue by parmentelat** [parmentelat/kube-install#16]
our use case is that the worker nodes from the FIT side will come and go

here's an attempt at formalising the expected behaviour

# reference state

for the sake of this discussion, our reference state is a running k8s cluster with its leader on the wired side - say sopnode-l1 - with one wireless node - say fit01 - belonging in the cluster

# scenarii

so from that point, the following may happen

## smooth cleanup

from the reference state:

* leader does a cleanup
  ```bash
  kubectl drain fit01
  kubectl delete fit01
 ```

* node does a cleanup
  ```bash
  kube-install.sh leave-cluster
  ```

when node joins the same cluster again, all should go well

## smooth cleanup + reload

from the reference state:

* leader does a cleanup
  ```bash
  kubectl drain fit01
  kubectl delete fit01
  ```

* node gets wiped off (re-imaged)

when node joins the same cluster again, all should go well

## impolite leave

from the reference state:

* the node gets re-imaged, and no cleanup is made on the cluster side

what I observe with v0.17 is that even in this case, the node joins successfully, and pre-existing pods get re-created on that node

it is unclear if that is something that users can rely on; my test was re-joining after about only 5 minutes, it might make sense to see if this still works with a longer delay

# conclusion

in short, I have not been able to reproduce the issue that some users say they are facing when recycling nodes

**Comment by parmentelat** [comment]
there is now a semi-automated test that checks for the smooth cleanup scenario, with

`all-in-one.sh setup leave-join`
